apps/ejabberd/src/gen_mod.erl

```diff
@@ -50,9 +50,9 @@
 
 -include("ejabberd.hrl").
 
--record(ejabberd_module, {module_host :: ejabberd:server(),
-                          opts :: list()
-                         }).
+-record(ejabberd_module, {module_host, opts}).
+-type ejabberd_module() :: #ejabberd_module{module_host :: {module(), ejabberd:server()},
```

Comment:
Fair enough. Although I used to add `| atom()` to such record definitions with comment "it's for match spec".